modules/ml/src/main/java/org/apache/ignite/ml/util/Utils.java

```diff
@@ -57,4 +58,29 @@
 
         return (T)obj;
     }
+
+    /**
+     * Select k distinct integers from range [0, n).
+     * @param n Number specifying left end of range of integers to pick values from.
+     * @param k Count specifying how many integers should be picked.
+     * @return Array containing k distinct integers from range [0, n);
+     */
+    public static int[] selectKDistinct(int n, int k) {
```

Comment:
This is not uniform distribution IMHO, please share the proof with dev-list

Reply:
It is https://en.wikipedia.org/wiki/Reservoir_sampling, check for example the section 'Algorithm in R', there is written that each number has equal probability to be chosen.